lib/ApiAuthenticationMiddleware.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * This file is part of RSS-Bridge, a PHP project capable of generating RSS and
+ * Atom feeds for websites that don't have one.
+ *
+ * For the full license information, please view the UNLICENSE file distributed
+ * with this source code.
+ *
+ * @package Core
+ * @license http://unlicense.org/ UNLICENSE
+ * @link    https://github.com/rss-bridge/rss-bridge
+ */
+
+final class ApiAuthenticationMiddleware
+{
+    public function __invoke($request): void
+    {
+        $accessTokenInConfig = Configuration::getConfig('authentication', 'access_token');
+        if (!$accessTokenInConfig) {
+            $this->exit('Access token is not set in this instance', 403);
+        }
+
+        if (isset($request['access_token'])) {
+            $accessTokenGiven = $request['access_token'];
+        } else {
+            $header = trim($_SERVER['HTTP_AUTHORIZATION'] ?? '');
+            $position = strrpos($header, 'Bearer ');
+
+            if ($position !== false) {
+                $accessTokenGiven = substr($header, $position + 7);
+            } else {
+                $accessTokenGiven = '';
+            }
+        }
+
+        if (!$accessTokenGiven) {
+            $this->exit('No access token given', 403);
```

Comment:
401 Unauthorized is the status code to return when the client provides no credentials or invalid credentials. 403 Forbidden is the status code to return when a client has valid credentials but not enough privileges to perform an action on a resource.